bolt/lib/Target/X86/X86MCPlusBuilder.cpp

```diff
@@ -1948,16 +1959,26 @@ class X86MCPlusBuilder : public MCPlusBuilder {
         std::optional<X86MemOperand> MO = evaluateX86MemoryOperand(Instr);
         if (!MO)
           break;
-        if (!isIndexed(*MO, R1))
-          // POSSIBLE_PIC_JUMP_TABLE
+        if (isRIPRel(*MO))
+          IsFixedBranch = true;
+        else if (isIndexed(*MO, R1))
+          ; // POSSIBLE_PIC_JUMP_TABLE
```

Comment:
In this case shouldn't we enforce that the MOV loading R1 actually appears after the LEA that defines R1?